build_cmake/configs/compile_test/pthread_cond_monotonic_test.c

```diff
@@ -0,0 +1,32 @@
+#include <errno.h>
```

Comment:
Out of curiosity: will this test be hooked into our automated testing system (and how), or it needs to be executed manually?

Reply:
Thanks for bringing this up! I just realized the build process only compiles the test, where as autoconf originally depended on the test successfully running before defining the CPP macro. I've just added an update that also check if it ran successfully.

This test gets built and run during the build configuration process (defined [here](https://github.com/wiredtiger/wiredtiger/blob/wt-7230-cmake-posix-x86/build_cmake/configs/auto.cmake#L277-L283)) to define the CPP macro `HAVE_PTHREAD_COND_MONOTONIC`. This being pretty much the same to how [autoconf](https://github.com/wiredtiger/wiredtiger/blob/develop/build_posix/configure.ac.in#L184) was testing it.
Every time the build is initialized (i.e. running `cmake ./..`), the test will be run and compiled. This _in a way_ being automated with every build. 
Should we make a more explicit hook into our automated testing system?

Reply:
Alright, thanks for the answer. I don't think a more explicit hook (to automated testing) is needed if the build initialization step is able to trigger the test.